### PR TITLE
CriticalSection Class Update (MERGE version - mbed-os-5.8)

### PR DESCRIPTION
**Pull request by deepikabhavnani**
CriticalSection Class was update and new member functions were added. See https://jira.arm.com/browse/IOTMORF-1773 and https://github.com/ARMmbed/mbed-os/pull/5621 for more info.

### CODE REVIEW

File: docs/reference/api/platform/CriticalSectionLock.md [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
+The CriticalSectionLock class provides a mechanism to access a resource without interruption. With the `CriticalSectionLock::enable` API, you can enter critical section with interrupts disabled. The `CriticalSectionLock::disable()` API is the exit from critical section, and the last exit call restores the state of interrupts.
 
 CriticalSectionLock class is based on RAII approach. In other words, the constructor acquires the lock, and the destructor destroys it automatically when it is out of scope. We do not recommend you use CriticalSectionLock as global or a member of a class because you will enter critical section on object creation, and all interrupts will be disabled.
 
+Nesting of critical section is supported, and the destructor enables interrupts only when you exit from the last nested critical section.
```
Comment (AnotherButler) [inline docs/reference/api/platform/CriticalSectionLock.md]: Query: Would it be correct to say "Mbed OS supports nesting of critical section, and the destructor..." for active voice?

Comment (deepikabhavnani) [reply docs/reference/api/platform/CriticalSectionLock.md]: Yes

**Comment by AnotherButler** [comment]
@deepikabhavnani Is it OK if I merge this into new_engine for 5.8, as long as I don't merge it into 5.7? Or should I wait?

**Comment by deepikabhavnani** [comment]
Yes

**Pull request merged**